app/models/proxy_config.rb

```diff
@@ -35,21 +35,20 @@ class InvalidEnvironmentError < StandardError; end
   scope :newest_first,   -> { order(version: :desc) }
   scope :by_environment, ->(env) { where(environment: VALID_ENVIRONMENTS[env]) }
   scope :by_host,        ->(host) { where.has { hosts =~ "%|#{host}|%" } if host }
-
-  scope :current_versions, -> do
-    table = BabySqueel[:proxy_configs].alias(:versions)
-    scope = joining { table.on((table.proxy_id == proxy_id) & (table.environment == environment)) }
-      .when_having { max(table.version) == version }
-      .group(:id)
-
-    System::Database.mysql? ? scope : where(id: scope.group(:version).select(:id))
-  end
-
-  scope :by_version, ->(version) do
-    next unless version
-    next current_versions if version == 'latest'
-
-    where(version: version)
+  scope :by_version,     ->(version) { where(version: version) if version }
+  scope :by_proxy_ids,   ->(proxy_ids) { where.has { proxy_id.in(proxy_ids) } if proxy_ids }
+
+  def self.latest_versions(environment:, proxy_ids: nil)
+    where.has do
```

Comment:
did you run `explain` on this SQL, and if so, can you share the result?

Reply:
I cannot share it publicly. It is in a comment in the Jira. Linked in the description of the PR.